Sources/StreamChatUI/Composer/ComposerVC_Tests.swift

```diff
@@ -0,0 +1,80 @@
+import SwiftUI
+import XCTest
+
+class ComposerVC_Tests: XCTestCase {
+    func testLevenshtein() throws {
+        XCTAssertEqual("".levenshtein(""), "".levenshtein(""))
+        XCTAssertEqual("".levenshtein(""), 0)
+        XCTAssertEqual("a".levenshtein(""), 1)
+        XCTAssertEqual("".levenshtein("a"), 1)
+        XCTAssertEqual("tommaso".levenshtein("ToMmAsO"), 4)
+    }
+
+    func testSearchByIDOrName() throws {
+        let user1 = ChatUser.mock(id: "searchingThis")
+        let user2 = ChatUser.mock(id: "x", name: "searchingThis")
+
+        XCTAssertEqual(searchUsers([user1], by: "sear"), [user1])
+        XCTAssertEqual(searchUsers([user2], by: "sear"), [user2])
+    }
+
+    func testMatchHappy() throws {
+        let tommaso = ChatUser.mock(id: "tommaso", name: "Tommaso")
+        let thierry = ChatUser.mock(id: "thierry", name: "Thierry")
+        let users = [tommaso, thierry]
+
+        XCTAssertEqual(searchUsers(users, by: "tom"), [tommaso])
+        XCTAssertEqual(searchUsers(users, by: "thier"), [thierry])
+    }
+
+    func testMatchHappyLowercased() throws {
+        let tommaso = ChatUser.mock(id: "tommaso", name: "Tommaso")
+        let thierry = ChatUser.mock(id: "thierry", name: "Thierry")
+        let users = [tommaso, thierry]
+
+        XCTAssertEqual(searchUsers(users, by: "Tom"), [tommaso])
+        XCTAssertEqual(searchUsers(users, by: "Thier"), [thierry])
+    }
+
+    func testMatchHappyTranslitterated() throws {
+        let tommaso = ChatUser.mock(id: "tommaso", name: "Tommaso")
+        let thierry = ChatUser.mock(id: "thierry", name: "Thierry")
+        let users = [tommaso, thierry]
+
+        XCTAssertEqual(searchUsers(users, by: "tóm"), [tommaso])
+        XCTAssertEqual(searchUsers(users, by: "thíer"), [thierry])
+    }
+
+    func testMatchHappySortedByDistance() throws {
+        let tommaso = ChatUser.mock(id: "tommaso", name: "tommaso")
+        let tomas = ChatUser.mock(id: "tomas", name: "tomas")
+        let users = [tommaso, tomas]
+
+        XCTAssertEqual(searchUsers(users, by: "tom"), [tomas, tommaso])
+    }
+
+    func testMatchHappyCyrillic() throws {
+        let petyo = ChatUser.mock(id: "42", name: "Петьо")
+        let anastasia = ChatUser.mock(id: "13", name: "Анастасiя")
+        let dmitriy = ChatUser.mock(id: "99", name: "Дмитрий")
+        let users = [petyo, anastasia, dmitriy]
+
+        XCTAssertEqual(searchUsers(users, by: "petyo"), [])
+        XCTAssertEqual(searchUsers(users, by: "Пе"), [petyo])
+        XCTAssertEqual(searchUsers(users, by: "Ана"), [anastasia])
+        XCTAssertEqual(searchUsers(users, by: "Дмитрии"), [dmitriy])
+        XCTAssertEqual(searchUsers(users, by: "Дмитри"), [dmitriy])
+    }
+
+    func testMatchHappyFrench() throws {
+        let user = ChatUser.mock(id: "fra", name: "françois")
+
+        XCTAssertEqual(searchUsers([user], by: "françois"), [user])
+        XCTAssertEqual(searchUsers([user], by: "franc"), [user])
+    }
```

Comment:
All these tests should be part of the UserSearchController. As for the testing name conventions, we should do:
`test_given_when_then`, where given is the function name. So for this case:
`test_searchUsers_whenHappyFrench_findsUser`.. something like this